File: mutuelleApi/models/Credit.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace mutuelleApi.models
{
    public class Credit : BaseEntity
    {
        public int MembreId { get; set; }
        public int Duree { get; set; }
        public double MontantCapital { get; set; }
        public double MontantCommission { get; set; }
        public double MontantInterets { get; set; }
        public string DateDemande { get; set; } = string.Empty;
        public string DateDecaissement { get; set; } = string.Empty;

        public List<Echeance> Echeances { get; set; } = [];
        public List<Mouvement>? Mouvements { get; set; }
        public Membre? Membre { get; set; }


        public double MontantTotal => MontantCapital + MontantCommission + MontantInterets;
        public int NombreEcheancePaye => Echeances?.Count(e => e.MontantCapitalRestant == 0)??0;
		public int NombreEcheanceImpaye => Echeances?.Count(e => e.MontantCapitalRestant > 0)??Duree;
		public string DateDerniereEcheance => Echeances?.Max(e => e.DateEcheance)??"";
        public double MontantCapitalRestant => Echeances?.Sum(m => m.MontantCapitalRestant) ?? 0;
        public string Status => NombreEcheanceImpaye == 0 ? "Remboursée" : Mouvements is not null && Mouvements.Count == 0 ? "Validée" : Mouvements is not null && Mouvements.Any(x => x.MontantDebit == MontantCapital) && NombreEcheancePaye == 0 ? "Décaissée" : "En cours";
        public string Nom => Membre?.Nom ?? "";
        public string NomSexe => Membre?.NomSexe ?? "";
        public string Photo => Membre?.Photo ?? "";
        public string NomAgence => Membre?.NomAgence ?? "";
        public int AgenceId  => Membre?.AgenceId ?? 0;
        public int CreditId => Id;

        [ForeignKey("ModifiePar")]
		public Utilisateur? Utilisateur { get; set; }
        public string UtilisateurLogin => Utilisateur?.Login ?? "";
    }
}

File: mutuelleApp/src/app/core/models/credit.ts
export class Credit {
  id?: number = 0;
  membreId?: number = 0;
  duree?: number = 0;
  montantCapital?: number = 0;
  montantCommission?: number = 0;
  montantInterets?: number = 0;
  dateDemande?: string = '';
  dateDecaissement?: string = '';

  montantTotal?: number = 0;
  montantCapitalRestant?: number = 0;
  montantCommissionRestant?: number = 0;
  montantInteretsRestant?: number = 0;

  status?: string = '';
  nombreEcheancePaye?: number = 0;
  nombreEcheanceImpaye?: number = 0;
  dateDerniereEcheance?: string = '';

  nom?: string = '';
  photo?: string = '';
  nomSexe?: string = '';
  agenceId?: number = 0;
}
